advanced/Templates/pihole-FTL.systemd

```diff
@@ -0,0 +1,36 @@
+[Unit]
+Description=Pi-hole FTL
+# This unit is supposed to indicate when network functionality is available, but it is only
+# very weakly defined what that is supposed to mean, with one exception: at shutdown, a unit
+# that is ordered after network-online.target will be stopped before the network
+Wants=network-online.target
+After=network-online.target
+# A target that should be used as synchronization point for all host/network name service lookups.
+# All services for which the availability of full host/network name resolution is essential should
+# be ordered after this target, but not pull it in.
+Wants=nss-lookup.target
+Before=nss-lookup.target
+
+[Service]
+User=pihole
+PermissionsStartOnly=true
+
+ExecStartPre=/opt/pihole/pihole-FTL-prestart.sh
+ExecStart=/usr/bin/pihole-FTL -f
+Restart=on-abnormal
+RestartSec=30s
```

Comment:
```suggestion
Restart=on-failure
RestartSec=5s
```

We should definitely use the `on-failure` policy here to at least *try* to recover from crashes, this isn't the case with `on-abnormal`:
![Screenshot from 2022-09-19 04-41-27](https://user-images.githubusercontent.com/16748619/190941922-b75f2344-45ab-48ea-a026-2a2b3fd37cbb.png)

> Setting this to `on-failure` is the recommended choice for long-running services, in order to increase reliability by attempting automatic recovery from errors.
> [source](https://www.freedesktop.org/software/systemd/man/systemd.service.html)

This will also make https://github.com/pi-hole/FTL/pull/1377 obsolete which is what triggered(/revived?) the whole idea here.


With `Restart=on-failure` auto-recovery from crash works fine:
![Screenshot from 2022-09-19 04-52-00](https://user-images.githubusercontent.com/16748619/190942710-593d0fa6-480f-443f-b00f-494993c0e85b.png)

Reply:
Makes sense: https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=

With 5 seconds delay before restarts, the start loop limit interval needs to be increased: https://www.freedesktop.org/software/systemd/man/systemd.unit.html#StartLimitIntervalSec=interval
It's 100ms by default, so Pi-hole could enter an endless restart loop. I'll increase it to 60 seconds so that it aborts after 5 attempts (default) up to 12 seconds each.